src/components/pagination/Pagination.tsx

```diff
@@ -0,0 +1,146 @@
+`;
+
+const StyledSvgIcon = styled(SvgIcon)<{ disabled?: boolean }>`
+  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
+`;
+
+const Pagination: FC<PaginationProps> = ({
+  totalItems,
+  itemsPerPage,
+  onPageChange,
+  currentPage,
+  label = "documents",
+  perPageOptions = [10, 25, 50, 100],
+  width = "100%",
+  handlePerPageChange,
+}) => {
+  const lastPage = Math.ceil(totalItems / itemsPerPage) - 1;
+  const isNextDisabled = currentPage === lastPage;
+  const isPrevDisabled = currentPage === 0;
+
+  const arrowFillColor = (isDisabled: boolean) =>
+    isDisabled ? "var(--neutral-200, #9ca3af)" : "var(--primary-300, #115873)";
+
+  return (
+    <PaginationContainer width={width}>
+      <ItemsPagesWrapper>
+        <PaginationWrapper>
+          <StyledSvgIcon
+            iconType={IconType.FIRST_PAGE}
+            onClick={() => onPageChange(0)}
+            fill={arrowFillColor(isPrevDisabled)}
+            disabled={isPrevDisabled}
+            data-testid="first-page-icon"
+          />
+          <ReactPaginate
+            className={styles["paginated-box"]}
+            breakLabel="..."
+            nextLabel={
+              <StyledSvgIcon
+                iconType={IconType.NEXT_ARROW}
+                onClick={() => !isNextDisabled && onPageChange(currentPage + 1)}
+                fill={arrowFillColor(isNextDisabled)}
+                disabled={isNextDisabled}
+                data-testid="next-page-icon"
+              />
+            }
+            previousLabel={
+              <StyledSvgIcon
+                iconType={IconType.BACK_ARROW}
+                onClick={() => !isPrevDisabled && onPageChange(currentPage - 1)}
+                fill={arrowFillColor(isPrevDisabled)}
+                disabled={isPrevDisabled}
+                data-testid="previous-page-icon"
+              />
+            }
+            pageRangeDisplayed={2}
+            pageCount={Math.ceil(totalItems / itemsPerPage)}
+            marginPagesDisplayed={1}
+            activeClassName={styles["ative-page"]}
+            disabledClassName={styles["disable-page"]}
+            pageClassName={styles["pages-count"]}
+            onPageChange={(data) => onPageChange(data.selected)}
+            previousClassName={styles["next-pre-btn"]}
+            nextClassName={styles["next-pre-btn"]}
+            forcePage={currentPage}
+            initialPage={currentPage}
+          />
+          <StyledSvgIcon
+            iconType={IconType.LAST_PAGE}
+            onClick={() => onPageChange(lastPage)}
+            fill={arrowFillColor(isNextDisabled)}
+            disabled={isNextDisabled}
+            data-testid="last-page-icon"
+          />
+        </PaginationWrapper>
+      </ItemsPagesWrapper>
+      <PerPageContainer>
+        <Dropdown
+          options={perPageOptions}
+          type="select"
+          onSelect={handlePerPageChange}
+          renderOption={(option) => option?.toString()}
```

Comment:
Why do we need to safely navigate here ?
```suggestion
          renderOption={(option) => option.toString()}
```